### PR TITLE
Update to Sync 2.1

### PR DESCRIPTION
**Pull request by cmelchior**

### CODE REVIEW

File: src/sync/sync_session.cpp [hunk 1]
```diff
@@ -540,11 +540,16 @@ void SyncSession::handle_error(SyncError error)
...
+            case ProtocolError::bad_client_version:
+            case ProtocolError::illegal_realm_path:
+            case ProtocolError::no_such_realm:
+            case ProtocolError::bad_changeset:
 #if REALM_SYNC_VER_MAJOR > 1
             case ProtocolError::bad_changeset_header_syntax:
             case ProtocolError::bad_changeset_size:
             case ProtocolError::bad_changesets:
             case ProtocolError::bad_decompression:
+            case ProtocolError::partial_sync_disabled:
```
Comment (simonask) [inline src/sync/sync_session.cpp]: Is it really on purpose that all these errors are ignored?

Comment (cmelchior) [reply src/sync/sync_session.cpp]: They are not being ignored, the session will still report the error. This switch just does special handling in some cases.

File: src/sync/sync_session.cpp [hunk 2]
```diff
@@ -540,11 +540,16 @@ void SyncSession::handle_error(SyncError error)
             case ProtocolError::reuse_of_session_ident:
             case ProtocolError::bound_in_other_session:
             case ProtocolError::bad_message_order:
+            case ProtocolError::bad_client_version:
```
[the remark this answers is not in the document]
Comment (cmelchior) [reply src/sync/sync_session.cpp]: They were outside the if before, so I don't think so?

File: src/sync/sync_session.cpp [hunk 3]
```diff
@@ -575,25 +580,18 @@ void SyncSession::handle_error(SyncError error)
                     user_to_invalidate->invalidate();
                 break;
             }
```
[the remark this answers is not in the document]
Comment (cmelchior) [reply src/sync/sync_session.cpp]: Good point, not sure what the idea is there. There seem to CI support for building both 1.* and 2.*, so I would probably do that in another PR?

[the remark this answers is not in the document]
Comment (beeender) [reply src/sync/sync_session.cpp]: yeah, i guess just create an issue for removing `REALM_SYNC_VER_MAJOR` and commit this.

**Pull request merged**